app/models/race_entry.rb

```diff
@@ -3,21 +3,18 @@
 class RaceEntry < ActiveRecord::Base
   belongs_to :racer
   belongs_to :race_edition
-     
+
   validates :racer, presence: true
   validates :race_edition, presence: true
-
+  validates_uniqueness_of :racer_id, scope: :race_edition_id,
+                          message: 'may be added to a race_edition only once'
 
   def elapsed_time
-    if time? then
-      ChronicDuration.output(time)
-    else
-      'NA'
-    end
+    time && ChronicDuration.output(time) || 'NA'
   end
-  
-  def set_elapsed_time(time_string)
-    time = ChronicDuration.parse(time_string)
+
+  def elapsed_time=(time_string)
+    time_string ||= ''
+    self.time = ChronicDuration.parse(time_string)
   end
```

Comment:
By renaming this method in this way, we create a "virtual attribute" on the RaceEntry model. Now we can (for example) pass an `elapsed_time` parameter from a form and the `time` attribute will be automatically set.